spring-cloud-starter-aws-secrets-manager-config/src/main/java/io/awspring/cloud/autoconfigure/secretsmanager/AwsSecretsManagerConfigDataLocationResolver.java

```diff
@@ -86,23 +88,36 @@ public boolean isResolvable(ConfigDataLocationResolverContext context, ConfigDat
 
 		AwsSecretsManagerPropertySources propertySources = new AwsSecretsManagerPropertySources(properties, log);
 
-		List<String> contexts = location.getValue().equals(PREFIX)
-				? propertySources.getAutomaticContexts(profiles.getAccepted())
-				: getCustomContexts(location.getNonPrefixedValue(PREFIX));
-
 		List<AwsSecretsManagerConfigDataResource> locations = new ArrayList<>();
-		contexts.forEach(
-				propertySourceContext -> locations.add(new AwsSecretsManagerConfigDataResource(propertySourceContext,
-						location.isOptional(), propertySources)));
+		if (location.getValue().equals(PREFIX)) {
+			List<String> contexts = propertySources.getAutomaticContexts(profiles.getAccepted());
+			contexts.forEach(propertySourceContext -> locations.add(new AwsSecretsManagerConfigDataResource(
+					propertySourceContext, location.isOptional(), propertySources)));
+			return locations;
+		}
+
+		Map<String, Boolean> mapOfLocation = getCustomContexts(location.getNonPrefixedValue(PREFIX));
+		mapOfLocation.forEach((variable, optional) -> locations
+				.add(new AwsSecretsManagerConfigDataResource(variable, optional, propertySources)));
 
 		return locations;
 	}
 
-	private List<String> getCustomContexts(String keys) {
+	private Map<String, Boolean> getCustomContexts(String keys) {
+		String optionalString = "optional";
+		Map<String, Boolean> mapOfValuesWithOptional = new HashMap<>();
 		if (StringUtils.hasLength(keys)) {
-			return Arrays.asList(keys.split(";"));
+			List<String> listOfFields = Arrays.asList(keys.split(";"));
+			listOfFields.forEach(field -> {
+				if (field.length() > 8 && field.toLowerCase().substring(0, 8).equals(optionalString)) {
```

Comment:
This gets too low level. I think we could have a dedicated class `CustomContext` that would take a string as an argument, and have methods like `isOptional`, `getName`. Then `getCustomContexts` would return just a list of these objects.

If its not clear enough, we can pair on this one.